### PR TITLE
[libc][NFC] refactor fmin and fmax

### PR DESCRIPTION
**Pull request by ghost**
Hello,

So, I worked on the fmaximum and fminimum functions recently and the reviewers suggested the structure:

```
if (bitsx ...)
  return ...;
if (bitsy ..)
  return 
...
return ...;
```
So I went ahead and did the same for fmin and fmax. I hope this isnt an issue for you all. thanks.

### CODE REVIEW

**Review by jhuber6 — COMMENTED** [review]
This is correct according to the LLVM coding standards so I'm in favor. I'll let some other `libc` people comment on whether or not `libc` adheres to that.

**Comment by Jobhdez** [comment]
> This is correct according to the LLVM coding standards so I'm in favor. I'll let some other `libc` people comment on whether or not `libc` adheres to that.

sounds good

**Review by lntue — APPROVED** [review]
Thanks for cleaning this up!

**Comment by Jobhdez** [comment]
> Thanks for cleaning this up!

no problem!

File: libc/src/__support/FPUtil/BasicOperations.h [hunk 1]
```diff
@@ -30,36 +30,32 @@ template <typename T, cpp::enable_if_t<cpp::is_floating_point_v<T>, int> = 0>
...
-  } else if (bitx.sign() != bity.sign()) {
+  if (bitx.sign() != bity.sign())
     // To make sure that fmax(+0, -0) == +0 == fmax(-0, +0), whenever x and
     // y has different signs and both are not NaNs, we return the number
     // with positive sign.
     return (bitx.is_neg() ? y : x);
-  } else {
-    return (x > y ? x : y);
-  }
+  return (x > y ? x : y);
```
Comment (nickdesaulniers) [inline libc/src/__support/FPUtil/BasicOperations.h:58]: Consider removing the parens from this line (and L42).

**Review by nickdesaulniers — APPROVED** [review]

File: libc/src/__support/FPUtil/BasicOperations.h (hunk 1 above)
Comment (Jobhdez) [reply libc/src/__support/FPUtil/BasicOperations.h:58]: Ok

Comment (jhuber6) [inline libc/src/__support/FPUtil/BasicOperations.h:58]:
```suggestion
  return x > y ? x : y;
```

File: libc/src/__support/FPUtil/BasicOperations.h [hunk 2]
```diff
@@ -30,36 +30,32 @@ template <typename T, cpp::enable_if_t<cpp::is_floating_point_v<T>, int> = 0>
...
-  } else if (bitx.sign() != bity.sign()) {
+  if (bitx.sign() != bity.sign())
     // To make sure that fmin(+0, -0) == -0 == fmin(-0, +0), whenever x and
     // y has different signs and both are not NaNs, we return the number
     // with negative sign.
     return (bitx.is_neg()) ? x : y;
-  } else {
-    return (x < y ? x : y);
-  }
+  return (x < y ? x : y);
```
Comment (jhuber6) [inline libc/src/__support/FPUtil/BasicOperations.h:42]:
```suggestion
  return x < y ? x : y;
```

**Review by jhuber6 — APPROVED** [review]
This stuff is easy with suggestions.

**Review by nickdesaulniers — APPROVED** [review]

File: libc/src/__support/FPUtil/BasicOperations.h [hunk 3]
```diff
@@ -30,36 +30,32 @@ template <typename T, cpp::enable_if_t<cpp::is_floating_point_v<T>, int> = 0>
...
     return y;
-  } else if (bity.is_nan()) {
+  if (bity.is_nan())
     return x;
-  } else if (bitx.sign() != bity.sign()) {
+  if (bitx.sign() != bity.sign())
     // To make sure that fmax(+0, -0) == +0 == fmax(-0, +0), whenever x and
     // y has different signs and both are not NaNs, we return the number
     // with positive sign.
     return (bitx.is_neg() ? y : x);
```
Comment (nickdesaulniers) [inline libc/src/__support/FPUtil/BasicOperations.h:57]: The difference in how the parentheses are used on L59 vs L41 could also be cleaned up (by omitting parentheses entirely).

**Review by nickdesaulniers — APPROVED** [review]

**Pull request merged**